File: frontend/src/types/immich.ts
export type ImmichImage = {
  immich_asset_id: string
  original_file_name: string
  file_created_at: string | null
  width: number | null
  height: number | null
  is_favorite: boolean
  thumbnail_url: string
}

export type ImmichImagePage = {
  items: ImmichImage[]
  total: number
  page: number
  page_size: number
  pages: number
}

export type ImmichAlbum = {
  immich_album_id: string
  album_name: string
  asset_count: number
  thumbnail_asset_id: string | null
  thumbnail_url: string | null
  start_date: string | null
  end_date: string | null
}

export type ImmichAlbumList = {
  items: ImmichAlbum[]
}

export type ImmichAssetLink = ImmichImage & {
  id: string
  asset_id: string
  created_at: string
  updated_at: string
}

export type ImmichAssetLinkList = {
  items: ImmichAssetLink[]
}

export type ImmichImageQuery = {
  page?: number
  page_size?: number
  search?: string
  album_id?: string
  favorite_only?: boolean
  taken_after?: string
  taken_before?: string
}

File: backend/app/schemas/immich.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ImmichImageRead(BaseModel):
    immich_asset_id: UUID
    original_file_name: str
    file_created_at: datetime | None = None
    width: int | None = Field(default=None, gt=0)
    height: int | None = Field(default=None, gt=0)
    is_favorite: bool
    thumbnail_url: str


class ImmichImagePageRead(BaseModel):
    items: list[ImmichImageRead]
    total: int = Field(ge=0)
    page: int = Field(ge=1)
    page_size: int = Field(ge=1)
    pages: int = Field(ge=0)


class ImmichAlbumRead(BaseModel):
    immich_album_id: UUID
    album_name: str
    asset_count: int = Field(ge=0)
    thumbnail_asset_id: UUID | None = None
    thumbnail_url: str | None = None
    start_date: datetime | None = None
    end_date: datetime | None = None


class ImmichAlbumListRead(BaseModel):
    items: list[ImmichAlbumRead]


class ImmichLinkWrite(BaseModel):
    asset_id: UUID
    immich_asset_id: UUID


class ImmichLinkRead(BaseModel):
    id: UUID
    asset_id: UUID
    immich_asset_id: UUID
    original_file_name: str
    file_created_at: datetime | None = None
    width: int | None = Field(default=None, gt=0)
    height: int | None = Field(default=None, gt=0)
    is_favorite: bool
    thumbnail_url: str
    created_at: datetime
    updated_at: datetime


class ImmichLinkListRead(BaseModel):
    items: list[ImmichLinkRead]
